src/render.tsx

```diff
@@ -132,6 +132,19 @@ function updateWithAct(
     act(() => {
       renderer.update(wrap(component));
     });
+
+    // Rebind queries and helper only when root instance changes.
+    // This will not affect render calls using `wrapper` or
+    // `unstable_validateStringsRenderedWithinText` options as they will create
+    // a wrapping root that does not change.
+    if (screen.container !== renderer.root) {
```

Comment:
When commenting this and adding the option unstable_validateStringsRenderedWithinText in the test it fails so I think the Profiler when using string validation does not prevent the root from changing.
 
Also if having a wrapper prevents the root from changing, could it be a solution to use some kind of default wrapper that would not change the queries behavior while still allow the root to stay the same ? (not sure at all if that's possible). If it worked it would probably cover both the use cases that currently fail

Reply:
Hmmm, good catch, I need to investigate it tomorrow.

Reply:
re default wrapper: that would be the easiest solution, and is certain terms this is what RTL is doing. However, there are serious downsides:
1. it's a breaking change as `container` will now point to the artificial root
2. all `toJSON()` tests will break, because of artificial root.

While 1. would be acceptable, and we could wait till next major release. The 2. issue is actually quite nasty, as we do not have focused `toJSON()` and can only call it on the whole render tree.